src/components/Header/UserPreferences/Automation/Decrypt.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { ReactElement, useEffect, useRef, useState } from 'react'
+import styles from './Decrypt.module.css'
+import Button from '../../../@shared/atoms/Button'
+import { toast } from 'react-toastify'
+import { useAutomation } from '../../../../@context/Automation/AutomationProvider'
+import Loader from '../../../@shared/atoms/Loader'
+import InputElement from '../../../@shared/FormInput/InputElement'
+
+export default function Decrypt(): ReactElement {
+  const {
+    isLoading,
+    decryptPercentage,
+    decryptAutomationWallet,
+    setIsAutomationEnabled
+  } = useAutomation()
+
+  const decrpytToastRef = useRef(null)
+  const passwordInputRef = useRef(null)
+
+  useEffect(() => {
+    toast.update(decrpytToastRef.current, { progress: decryptPercentage })
+  }, [decrpytToastRef, decryptPercentage])
```

Comment:
nitpick: The variable name decrpytToastRef seems to have a typo. It might be clearer if named decryptToastRef.